katdal/datasources.py

```diff
@@ -36,13 +36,14 @@
 from .sensordata import TelstateSensorData, TelstateToStr
 from .chunkstore_s3 import S3ChunkStore
 from .chunkstore_npy import NpyFileChunkStore
+from .chunkstore import StoreUnavailable
 from .flags import DATA_LOST
 
 
 logger = logging.getLogger(__name__)
 
 
-class DataSourceNotFound(Exception):
+class DataSourceNotFound(OSError):
```

Comment:
If you do this, you should pass an errno and strerror to the constructor when available (e.g. when catching a real OSError and raising). I'm not 100% sure it's a good idea - it's normally used for system calls.

Reply:
`StoreUnavailable` has been an `OSError` since the start... Also, all instances of `DataSourceNotFound` were raised after either an actual `OSError` or a `ConnectionError`, which is also an `OSError`. So I'm not 100% sure it's a bad idea :-)

Reply:
In that case it's probably a good idea, but we should pass through the (errno, strerror) pair from the underlying OSError.

Reply:
Gave up on this for now (since `ChunkNotFound` is a `KeyError` and not an `OSError`, it's all muddied anyway).